javaCreateAccount/src/main/java/com/hamdan/forzenbook/java/createaccount/JavaCreateAccountActivity.java

```diff
@@ -0,0 +1,244 @@
+                                    binding.emailErrorText.setVisibility(View.VISIBLE);
+                                }
+                            } else {
+                                binding.emailErrorText.setVisibility(View.GONE);
+                            }
+
+                            if (state.getLocation().getError().getState() == ErrorOutcomes.LENGTH) {
+                                binding.locationErrorText.setVisibility(View.VISIBLE);
+                            } else {
+                                binding.locationErrorText.setVisibility(View.GONE);
+                            }
+
+                        }
+                );
+
+        binding.inputFirstNameText.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                        model.updateCreateAccountTextAndErrors(
+                                charSequence.toString(),
+                                lastNameValue,
+                                birthDateValue,
+                                emailValue,
+                                locationValue
+                        );
+                    }
+
+                    @Override
+                    public void afterTextChanged(Editable editable) {
+                    }
+                }
+        );
+        binding.inputLastNameText.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                        model.updateCreateAccountTextAndErrors(
+                                firstNameValue,
+                                charSequence.toString(),
+                                birthDateValue,
+                                emailValue,
+                                locationValue
+                        );
+                    }
+
+                    @Override
+                    public void afterTextChanged(Editable editable) {
+                    }
+                }
+        );
+        binding.inputEmailText.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                        model.updateCreateAccountTextAndErrors(
```

Comment:
Make one function that does this and call that instead of repeating this big block